evm/src/fuzz/invariant/executor.rs

```diff
@@ -256,6 +275,99 @@ impl<'a> InvariantExecutor<'a> {
+    /// format). They will be used to filter contracts after the `setUp`, and more importantly,
+    /// during the runs.
+    ///
+    /// Priority:
+    ///
+    /// targetArtifactSelectors > excludeArtifacts > targetArtifacts
+    pub fn select_contract_artifacts(
+        &mut self,
+        invariant_address: Address,
+        abi: &Abi,
+    ) -> eyre::Result<()> {
+        // targetArtifactSelectors -> (string, bytes4[])[].
+        let targeted_abi = self
+            .get_list::<(String, Vec<FixedBytes>)>(
+                invariant_address,
+                abi,
+                "targetArtifactSelectors",
+            )
+            .into_iter()
+            .map(|(contract, functions)| (contract, functions))
+            .collect::<BTreeMap<_, _>>();
+
+        // Insert them into the executor `targeted_abi`.
+        for (contract, selectors) in targeted_abi {
+            let identifier = self.validate_selected_contract(contract, &selectors)?;
+
+            self.targeted_abi.entry(identifier).or_insert(vec![]).extend(selectors);
+        }
+
+        // targetArtifacts -> string[]
+        // excludeArtifacts -> string[].
+        let [selected_abi, excluded_abi] = ["targetArtifacts", "excludeArtifacts"]
+            .map(|method| self.get_list::<String>(invariant_address, abi, method));
+
+        // Insert `excludeArtifacts` into the executor `excluded_abi`.
+        for contract in excluded_abi {
+            let identifier = self.validate_selected_contract(contract, &[])?;
+
+            if !self.excluded_abi.contains(&identifier) {
+                self.excluded_abi.push(identifier);
+            }
+        }
+
+        // Insert `targetArtifacts` into the executor `targeted_abi`, if they have not been seen
+        // before.
+        for contract in selected_abi {
+            let identifier = self.validate_selected_contract(contract, &[])?;
+
+            if !self.targeted_abi.contains_key(&identifier) &&
+                !self.excluded_abi.contains(&identifier)
+            {
+                self.targeted_abi.insert(identifier, vec![]);
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Makes sure that the contract exists in the project. If so, it returns its artifact
+    /// identifier.
+    fn validate_selected_contract(
+        &mut self,
+        contract: String,
+        selectors: &[FixedBytes],
+    ) -> eyre::Result<String> {
+        let contracts = self
+            .project_contracts
+            .iter()
+            .filter(|(artifact, _)| {
+                // Check if user passed a matching contract name or source:name
+                artifact.name.as_str() == contract || artifact.identifier() == contract
+            })
+            .collect::<Vec<_>>();
+
+        if contracts.len() > 1 {
+            eyre::bail!("{contract} has more than one implementation. Please specify it with `contract_path:contract_name`.");
+        }
+
```

Comment:
We should probably refactor this to a wrapper type instead of a map and add a `find(contract: &str) -> Contract`  method